### PR TITLE
feat(LINGUIST-270): Message count stats implemented

### PR DESCRIPTION
**Pull request by KardelenCeren**
Stats related to the number of messages a user sent over time.

### CODE REVIEW

File: chat/tests.py [hunk 1]
```diff
@@ -1,3 +1,102 @@
 from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+from .models import ChatBot, Conversation, Message
+from constants.header_constants import HEADER_USER_EMAIL
 
-# Create your tests here.
+
```
Comment (can-aslan) [inline chat/tests.py:8]: It seems like there are two new lines here, if so please reduce to one

File: chat/views/get_message_count_aggregate.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+from rest_framework.decorators import api_view
+from chat.models import Message
+from constants.header_constants import HEADER_USER_EMAIL
+from utils.http_utils import generate_error_response, generate_success_response
+from drf_yasg import openapi
+from drf_yasg.utils import swagger_auto_schema
+from django.db.models import Count
+from django.db.models import F
+
+
```
Comment (can-aslan) [inline chat/views/get_message_count_aggregate.py:12]: It seems like there are two new lines here, if so please reduce to one

File: chat/views/get_message_count_aggregate.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                "application/json": {
+                    "timestamp": "2021-08-30 14:00:00",
+                    "status": 400,
+                    "msg": "Authentication is required"
+                }
+            }
+        )
+    }
+)
+@api_view(['GET'])
```
Comment (can-aslan) [inline chat/views/get_message_count_aggregate.py:66]: In `get_message_count_by_bot.py`, there is a new line between `)` and `@api_view(['GET'])`. However, here, there is no new line. I don't know the best practice here, but please select one of the two for consistency, thanks 😄

File: chat/views/get_message_count_by_bot.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                    "timestamp": "2021-08-30 14:00:00",
+                    "status": 400,
+                    "msg": "Authentication is required"
+                }
+            }
+        )
+    }
+)
+
+@api_view(['GET'])
```
Comment (can-aslan) [inline chat/views/get_message_count_by_bot.py:76]: In `get_message_count_aggregate.py`, there is a no new line between `)` and `@api_view(['GET'])`. However, here, there is a new line. I don't know the best practice here, but please select one of the two for consistency, thanks 😄

File: chat/views/get_message_count_aggregate.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    # Get the value of num_days from the query parameters
+    num_days = request.GET.get('daysLimit', None)
+    if num_days is not None:
+        try:
+            num_days = int(num_days)
+        except ValueError:
+            return generate_error_response(400, "Invalid value for 'num_days'. Must be an integer.")
+
+    # Calculate the start date based on num_days
+    if num_days is not None:
```
Comment (can-aslan) [inline chat/views/get_message_count_aggregate.py]: Between lines `73` and `80`, `num_days is not None` is checked twice. It seems possible to merge the two checks into one, which would be better for performance and readability of the code

File: chat/views/get_message_count_by_bot.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    # Get the value of num_days from the query parameters
+    num_days = request.GET.get('daysLimit', None)
+    if num_days is not None:
+        try:
+            num_days = int(num_days)
+        except ValueError:
+            return generate_error_response(400, "Invalid value for 'num_days'. Must be an integer.")
+
+    # Calculate the start date based on num_days
+    if num_days is not None:
```
Comment (can-aslan) [inline chat/views/get_message_count_by_bot.py:109]: Between lines `92` and `99`, `num_days is not None` is checked twice. It seems possible to merge the two checks into one, which would be better for performance and readability of the code

**Review by can-aslan — CHANGES_REQUESTED** [review]
Looks good, just left some minor comments. Eline saglikkk 🚀 

Note: The title of the PR does not conform to our [PR Conventions](https://linguistai.atlassian.net/wiki/spaces/LINGUIST/pages/557057/Pull+Request+Conventions), please edit the title (just the title is fine)🙏

File: chat/views/get_message_count_by_bot.py (hunk 4 above)
Comment (KardelenCeren) [reply chat/views/get_message_count_by_bot.py:76]: Yes, there should not be one - fixed it.

File: chat/views/get_message_count_aggregate.py (hunk 3 above)
Comment (KardelenCeren) [reply chat/views/get_message_count_aggregate.py:66]: Yes, there should not be a new line there.

File: chat/views/get_message_count_aggregate.py (hunk 5 above)
Comment (KardelenCeren) [reply chat/views/get_message_count_aggregate.py]: Merged the two checks

File: chat/views/get_message_count_by_bot.py (hunk 6 above)
Comment (KardelenCeren) [reply chat/views/get_message_count_by_bot.py:109]: Merged the two checks

File: chat/tests.py (hunk 1 above)
Comment (KardelenCeren) [reply chat/tests.py:8]: That's a feature not a bug! :P  Python's convention is two new lines before class definitions. 
https://peps.python.org/pep-0008/#blank-lines

Comment (can-aslan) [reply chat/tests.py:8]: It is indeed! I didn't know this, thanks for sharing 😄

**Review by can-aslan — DISMISSED** [review]
LGTM

File: README.md [hunk 7]
```diff
@@ -65,3 +65,14 @@ In order to run the server:
 ```bash
 python manage.py runserver
 ```
+
+In order to run the tests:
+
+```bash
+pip install pytest-django
```
Comment (tolgaozgun) [inline README.md]: Bunu bence yine requirements'a ekleyelim. Soft-dependancy için ayrı bir sistemimiz olmadığı için hard-dependancy gibi düşünebiliriz

**Comment by tolgaozgun** [comment]
Onun dışında güzel duruyor eline sağlık

**Review by tolgaozgun — CHANGES_REQUESTED** [review]
Review'a bakabilir misin

File: README.md (hunk 7 above)
Comment (KardelenCeren) [reply README.md]: Yes doğru ekledim

**Review by tolgaozgun — APPROVED** [review]
LGTM

**Pull request merged**